### DISCUSSION TITLE
Exporting Whisper model in native format

### DISCUSSION THREAD

**Discussion by alumae** [espnet/espnet#5024] [General]
Can the Espnet-based finetuned Whisper model be exported in the native Whisper format? Whisper has some nice and fast C++ based decoders, it would be useful for many people. Also, models hosted on Huggingface could be directly used via the HF inference API.

**Comment by sw005320** [comment]
@alumae, this is a great suggestion.
@slSeanWU, can you consider this possibility?
You can directly discuss it with me internally or discuss it here.

**Comment by alumae** [comment]
I took a look at it myself and I think it's actually better to convert the Espnet model to HF transformers format which can be used by e.g. https://github.com/guillaumekln/faster-whisper. I actually implemented it myself (largely based on https://github.com/huggingface/transformers/blob/main/src/transformers/models/whisper/convert_openai_to_hf.py), it's available [here](https://gist.github.com/alumae/2dcf473b667cec9d513b80ea24e94672). The resulting model is available [here](https://huggingface.co/TalTechNLP/whisper-medium-et) and it seems to be fully compatible with HF.

**Comment by slSeanWU** [comment]
Hi @alumae, thanks for the suggestions.

Yeah, converting checkpoint from ESPnet to HF format seems to be feasible (and good in that we don't have to touch any infrastructure parts). However, this likely still involves quite some engineering efforts, for which I'm not sure if I can allocate time in the near term.

Would you be interested in implementing it yourself?

**Comment by alumae** [comment]
Yes, I already implemented the conversion script, see my comment above :)

**Comment by slSeanWU** [comment]
yeah I saw that, but the model formats of ESPnet whisper and openai-whisper are different

**Comment by alumae** [comment]
Yes, but I ported it to read Espnet models, see [here](https://gist.github.com/alumae/2dcf473b667cec9d513b80ea24e94672). Use:

    python convert_espnet_full_whisper_to_hf.py \
      --whisper-config-id openai/whisper-medium  \
      --espnet_checkpoint exp/asr_whisper_medium_finetune_lr1e-5_adamw_wd1e-2_3epochs/valid.acc.best.pth \
      --pytorch_dump_folder_path hf/whisper-medium-et

**Reply by slSeanWU** [reply]
oops, the links slipped through my eyes, sorry about that 😅. Thanks a lot for porting our model!
